app/controllers/api/distributions_controller.rb

```diff
@@ -14,4 +17,26 @@ def owner_resource
   def filtered(arel)
     polymorphic_filtered('distributable', arel)
   end
+
+  def create
+    create_resource.tap do |res|
+      consume! res, create_options
+      res = res.becomes(res.type.safe_constantize) if res.type
+      hal_authorize res
+      res.save!
+      distribute(res)
```

Comment:
yeah, wasn't sure this was worth refactoring yet, since I don't see other cases of this besides distributions.